### PR TITLE
Media Library #5: Search

### PR DESCRIPTION
**Pull request by frosty**
This PR adds search to the media library. The search will search the title, caption, and description fields of media items.

**Note:**

* There's a noticeable flickering when the collection view is refreshed as you type your search query. This is due to the way the media picker loads its images asynchronously, and will be fixed separately in the media picker.
* Search currently doesn't have a 'no results' state. I'll be adding no results separately for when a user has no items in their media library, so I'll address search at the same time.

**To test:**

* Try searching a media library!
* Try editing metadata for an item and check you can search for the new values
* Check the scroll insets play correctly with the keyboard

Needs review: @kurzee

### CODE REVIEW

**Comment by kurzee** [comment]
There's an interesting case in which:

1. Open Media, search for a known term matching a media item.
2. Select the media item.
3. Delete the media item.
4. Notice the controller pops and reset the root picker, but the search text is still present.
5. Also notice that trying to search again, with new text, does nothing.

File: WordPress/Classes/ViewRelated/Media/MediaLibraryPickerDataSource.m [hunk 1]
```diff
@@ -308,6 +308,23 @@ + (NSPredicate *)predicateForFilter:(WPMediaType)filter blog:(Blog *)blog
...
+
+    return nil;
+}
+
+- (void)setSearchQuery:(NSString *)searchQuery
+{
+    _searchQuery = [searchQuery copy];
+
+    _fetchController = nil;
+    [self.fetchController performFetch:nil];
```
Comment (kurzee) [inline WordPress/Classes/ViewRelated/Media/MediaLibraryPickerDataSource.m]: Would we want to wrap this in the usual `if (_searchQuery != searchQuery) { }`?

**Comment by frosty** [comment]
@kurzee Thanks for catching that! Pushed up some commits to fix the issues you mentioned – care to take another look?

**Comment by frosty** [comment]
Thanks @kurzee – had a brain failure moment there :)

**Pull request merged**